File: tasks (demian)/block 2/day 8/task 2/task 2.cpp
int t;
cin >> t;

while (t--) {
    int k;
    cin >> k;

    int MAX = -inf, MIN = inf;
    // a b -> b ( a + 1 ) + a (b + 1) = 2 ab + a + b = k -> k - a = 2ab + b -> b
    // = (k - a) / (2a + 1) k = 1e9 -> 2ab + a + b -> 2a ^ 2 + 2a = 1e9 -> a ~
    // 1e5

    for (int a = 1; a <= 100000; a++) {
        int b = (k - a) / (2 * a + 1);
        if (2 * a * b + a + b == k && b > 0) {
            MIN = min(MIN, a * b);
            MAX = max(MAX, a * b);
        }
    }

    if (MAX == -inf) {
        cout << -1 << '\n';
    } else {
        cout << MIN << " " << MAX << '\n';
    }
}

File: OPT/lomonosov/task 2/task 2.py
def convert(s : str) -> dict:
    d = dict()
    
    temp = 0
    for c in s:
        if c.isdigit():
            temp = 10 * temp + int(c)
        else:
            if c in d:
                d[c] += temp
            else:
                d[c] = temp
            temp = 0
    
    return d

costs = {
    'a' : 0.0001,
    'b' : 0.0005,
    'c' : 0.001,
    'd' : 0.005,
    'e' : 0.01,
    'f' : 0.02,
    'g' : 0.05,
    'h' : 0.1,
    'i' : 0.25,
    'A' : 0.05,
    'B' : 0.1,
    'C' : 0.5,
    'D' : 1,
    'E' : 2,
    'F' : 5,
    'G' : 10,
    'H' : 20,
    'I' : 50
}
def value(d : dict) -> float:
    global costs

    answer = 0
    for c in d:
        answer += d[c] * costs[c]
    
    return answer

n = int(input())
a = []
for _ in range(n):
    s = str(input())
    a.append(value(convert(s)))

mx = 0
ansi, ansj = 0, 0
for i in range(n):
    for j in range(n):
        if i != j:
            if abs(a[i] - a[j]) > mx:
                ansi = i
                ansj = j
                mx = abs(a[i] - a[j])
            if abs(a[i] - a[j]) == mx:
                if ansi + ansj < i + j:
                    ansi = i
                    ansj = j
                    mx = abs(a[i] - a[j])

print(min(ansi, ansj) + 1)
print(max(ansi, ansj) + 1)